### DISCUSSION TITLE
blocksqp silencing print (specifically with qpoases as the qp solver)

### DISCUSSION THREAD

**Discussion by sepsamavi** [casadi/casadi#3521] [Q&A] [answered]
Hi all, 

tldr: Is there any way to silence the qpoases licence printout every time the blocksqp is called? 

In Python, I am currently using nlpsol and blocksqp with qpoases as the qp solver. I have been unable to completely silence the printing from qpoases. 

Here is a snippet of how the solver is created, 

```
import casadi as cs

...

prob = ...
opts = {}
opts["print_time"] = 0
opts["calc_lam_p"] = False
opts["expand"] = True
opts["jit"] = True
opts["compiler"] = "shell"
opts["jit_options"] = {"verbose":True}
opts["qpsol"] = "qpoases"
opts["regularity_check"] = True
opts["qpsol_options"] = {"printLevel":"none", "enableRegularisation":True}
opts["max_iter"] = 10
opts["print_iteration"] = False
opts["print_header"] = False

solver = cs.nlpsol('solver_robws', 'blocksqp', prob, opts)

...

```

I have also added printLevel in qpsol_options following advice from previous community discussions regarding qpoases, however during the JIT process, the following warning indicates that these settings are ignored:

```
WARNING("Option 'qpsol' currently not supported, ignored") [.../casadi/interfaces/blocksqp/blocksqp.cpp:297]
WARNING("Option 'qpsol_options' currently not supported, ignored") [.../casadi/interfaces/blocksqp/blocksqp.cpp:300]
```

By setting print_iteration and print_header, most of the print statements are removed, however,
the following message is printed every time the solver is called: 

```
qpOASES -- An Implementation of the Online Active Set Strategy.
Copyright (C) 2007-2015 by Hans Joachim Ferreau, Andreas Potschka,
Christian Kirches et al. All rights reserved.

qpOASES is distributed under the terms of the 
GNU Lesser General Public License 2.1 in the hope that it will be 
useful, but WITHOUT ANY WARRANTY; without even the implied warranty 
of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
See the GNU Lesser General Public License for more details.
```

Several instances of similar solvers are called multiple times in each iteration of my problem and these print statements clog up the console. Is there any way I can prevent these messages from being printed?

Best, 
Sepehr

**Comment by jgillis** [comment, marked as answer]
I don't think it's possible without changing BlockSQP implementation.
A workaround is to capture and hide output from the Python side (e.g. https://stackoverflow.com/questions/2828953/silence-the-stdout-of-a-function-in-python-without-trashing-sys-stdout-and-resto).
